### PR TITLE
Set device tags on device create/details screens

### PR DESCRIPTION
**Pull request by audserraCGI**
## Description

What's new?

- #236 

## What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [x] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/DeviceTwinMapperTests.cs [hunk 1]
```diff
@@ -49,8 +49,18 @@ public void CreateDeviceDetails_StateUnderTest_ExpectedBehavior()
...
+            {
+                { "assetId", Guid.NewGuid().ToString() },
+                { "locationCode", Guid.NewGuid().ToString() }
+            };
+
+            foreach (KeyValuePair<string, string> customTag in customTags)
+            {
+                twin.Tags[customTag.Key.ToCamelCase()] = customTag.Value;
+            }
+
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/DeviceTwinMapperTests.cs:63]: Maybe a more elegant way to do that ? 
```suggestion
            twin.Tags["assetId"] = Guid.NewGuid().ToString();
            twin.Tags["locationCode"] = Guid.NewGuid().ToString();
```

File: src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/LoRaDeviceTwinMapperTests.cs [hunk 2]
```diff
@@ -40,8 +41,17 @@ public void CreateDeviceDetails_StateUnderTest_ExpectedBehavior()
...
+            Dictionary<string, string> customTags = new()
+            {
+                { "assetId", Guid.NewGuid().ToString() },
+                { "locationCode", Guid.NewGuid().ToString() }
+            };
+
+            foreach (KeyValuePair<string, string> customTag in customTags)
+            {
+                DeviceHelper.SetTagValue(twin, customTag.Key, customTag.Value);
+            }
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/LoRaDeviceTwinMapperTests.cs:54]: Same as previous

File: src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/LoRaDeviceTwinMapperTests.cs [hunk 3]
```diff
@@ -110,8 +123,17 @@ public void UpdateTwin_StateUnderTest_ExpectedBehavior()
...
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server.Tests.Unit/Mappers/LoRaDeviceTwinMapperTests.cs:136]: Idem

File: src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs [hunk 4]
```diff
@@ -60,30 +49,7 @@ public DeviceTagSettingsController(ILogger<DeviceTagSettingsController> log, IDe
         [HttpPost(Name = "POST a set of device tag settings")]
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs:49]:
```suggestion
        [HttpPost(Name = "POST Update the Device tags set")]
```

File: src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs [hunk 5]
```diff
@@ -94,27 +60,7 @@ await this.tableClientFactory
         [HttpGet(Name = "GET a set of device settings")]
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs:60]:
```suggestion
        [HttpGet(Name = "GET Device tags settings")]
```

File: src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs [hunk 6]
```diff
@@ -1,120 +1,66 @@
...
+            this.deviceTagService = deviceTagService;
         }
 
         /// <summary>
         /// Updates the device tag settings to be used in the application.
         /// </summary>
         /// <param name="tags">List of tags.</param>
         /// <returns>The action result.</returns>
-        [HttpPost(Name = "POST a set of device tag settings")]
+        [HttpPost(Name = "Update the Device tags set")]
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Server/Controllers/v1.0/DeviceTagSettingsController.cs:49]:
```suggestion
        [HttpPost(Name = "POST Update the Device tags settings")]
```

**Review by kbeaugrand — DISMISSED** [review]

**Review by Sben65 — APPROVED** [review]

**Pull request merged**